src/state/current-user-auth/reducer.js

```diff
@@ -1,11 +1,27 @@
 import { SET_CURRENT_USER_AUTH, CLEAR_CURRENT_USER_AUTH } from 'state/current-user-auth/types';
+import { get, flatten } from 'lodash';
 
-const reducer = (state = [], action = {}) => {
+const stringToArr = val => (
+  Array.isArray(val) ? val : [val]
+);
+
+const initialState = {
+  auth: [],
+  roles: [],
+};
+
+const reducer = (state = initialState, action = {}) => {
   switch (action.type) {
-    case SET_CURRENT_USER_AUTH:
-      return [...action.payload];
+    case SET_CURRENT_USER_AUTH: {
+      const { result, allPermissions } = action.payload;
+      const authMaps = result.map(auths => (
+        stringToArr(auths.permissions ? auths.permissions : get(auths, 'role', []))
+      ));
+      const roles = Array.from(new Set(flatten(authMaps)));
+      return { ...state, auth: result, roles: roles.includes('superuser') ? allPermissions : roles };
```

Comment:
Please refactor all the `superuser` occurrences to a constant.

Reply:
constant now under `state/permissions/const.js`